### ISSUE TITLE
space-infix-ops: false positive on "--" in template strings

### ISSUE THREAD

**Issue by rondonjon** [eslint/eslint#9634]
**Tell us about your environment**

* **ESLint Version:** 4.11.0
* **Node Version:** 8.9.1
* **npm Version:** 5.0.4

**What parser (default, Babel-ESLint, etc.) are you using?**

webpack, Babel

**Please show your full configuration:**

```js
...
"space-infix-ops": 1,
...
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I use webpack with eslint.

**What did you expect to happen?**

The `space-infix-ops` rule should ignore the (text) contents of ES6 template strings.
It should, in particular, not consider BEM CSS inside template strings as Javascript code.

**What actually happened? Please include the actual, raw output from ESLint.**

This string causes a `space-infix-ops` rule violation:

```
const className=`App App--variation${this.state.variation}`;
```

Apparently, the rule detects the `--` and falsely assumes that it is an infix operation on `App`.

**Comment by not-an-aardvark** [comment]
From testing out your code in the [demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IHNwYWNlLWluZml4LW9wczogZXJyb3IgKi9cblxuY29uc3QgY2xhc3NOYW1lPWBBcHAgQXBwLS12YXJpYXRpb24ke3RoaXMuc3RhdGUudmFyaWF0aW9ufWA7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6e319fQ==), it appears that a `space-infix-ops` error is being reported for the `=` symbol in the variable declaration, not the `--` in your string. Are you sure that the characters in the string were reported rather than the `=` symbol?

If so, could you please include the actual, raw output from ESLint?

**Comment by rondonjon** [comment]
Ouw, my bad :-( Thanks for pointing this out, and sorry for the hassle :-)
